core/domain/question_services.py

```diff
@@ -252,18 +257,27 @@ def get_questions_by_skill_ids(total_question_count, skill_ids):
             length of skill_ids, and it will be rounded up if not evenly
             divisible. If not enough questions for one skill, simply return
             all questions linked to it. The order of questions will follow the
-            order of given skill ids, but the order of questions for the same
-            skill is random.
+            order of given skill ids, and the order of questions for the same
+            skill is random when sort_by_mastery is false, otherwise the order
+            is sorted by the difference between skill difficulty and mastery.
     """
 
     if total_question_count > feconf.MAX_QUESTIONS_FETCHABLE_AT_ONE_TIME:
         raise Exception(
             'Question count is too high, please limit the question count to '
             '%d.' % feconf.MAX_QUESTIONS_FETCHABLE_AT_ONE_TIME)
 
-    question_skill_link_models = (
-        question_models.QuestionSkillLinkModel.get_question_skill_links_equidistributed_by_skill( #pylint: disable=line-too-long
-            total_question_count, skill_ids))
+    if sort_by_mastery:
+        degrees_of_mastery = skill_services.get_multi_user_skill_mastery(
+            user_id, skill_ids)
+        question_skill_link_models = (
+            question_models.QuestionSkillLinkModel.get_sorted_question_skill_links_equidistributed_by_skill( #pylint: disable=line-too-long
```

Comment:
get_question_skill_links_based_on_user_mastery_equidistributed_by_skill? Or something similar? Can't think of a smaller name

Reply:
Done.